Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -46,19 +48,22 @@ public final class RegistryClient {
     private let archiverProvider: (FileSystem) -> Archiver
     private let httpClient: HTTPClient
     private let authorizationProvider: HTTPClientAuthorizationProvider?
+    private let fingerprintStorage: PackageFingerprintStorage?
     private let jsonDecoder: JSONDecoder
 
     public init(configuration: RegistryConfiguration,
                 identityResolver: IdentityResolver,
                 customArchiverProvider: ((FileSystem) -> Archiver)? = nil,
                 customHTTPClient: HTTPClient? = nil,
-                authorizationProvider: HTTPClientAuthorizationProvider? = nil)
+                authorizationProvider: HTTPClientAuthorizationProvider? = nil,
+                fingerprintStorage: PackageFingerprintStorage? = nil)
```

Comment:
should authorizationProvider and fingerprintStorage be non-optional? they feel like required (ie they are not customization like customHTTPClient). If this is optional for tests, we can create extensions in the test module to make it easier to construct while keeping the main constructor "correct"

Reply:
The only reason `fingerprintStorage` is optional is because `sharedCacheDirectory` is optional. What should we default storage path to in that case?

Reply:
`fingerprintStorage` is non-optional https://github.com/apple/swift-package-manager/pull/3890/commits/036385bd54c43e6ae0ca70a83de74af417b33f68